com.unity.cinemachine/Editor/Editors/CinemachineSmoothPathEditor.cs

```diff
@@ -244,6 +247,142 @@ void DrawPositionControl(int i, Matrix4x4 localToWorld, Quaternion localRotation
             }
         }
 
+#if CINEMACHINE_USE_BATCH_POINT_LINE_APIS
+        static Vector4[] m_BezierWeights;
+        static int m_BezierWeightsResolution = -1;
+        static Vector3[] m_StepPoints;
+        static Vector3[] m_LeftRailPoints;
+        static Vector3[] m_RightRailPoints;
+        static Vector3[] m_SleeperPoints;
+
+        static ElementType[] EnsureArraySize<ElementType>(ElementType[] array, int requiredSize)
+        {
+            return ((array == null) || (array.Length < requiredSize)) ? new ElementType[requiredSize] : array;
+        }
+
+        [DrawGizmo(GizmoType.Active | GizmoType.NotInSelectionHierarchy
+             | GizmoType.InSelectionHierarchy | GizmoType.Pickable, typeof(CinemachineSmoothPath))]
+        static void DrawGizmos(CinemachineSmoothPath path, GizmoType selectionType)
+        {
+            int numSegments = path.m_Waypoints.Length - (path.Looped ? 0 : 1);
```

Comment:
Why is there a separate implementation for this?  Can't you just call CinemachinePathEditor.DrawPathGizmo?